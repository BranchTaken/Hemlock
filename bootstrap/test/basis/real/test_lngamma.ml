open! Basis.Rudiments
open! Basis
open Real
open Format

let test () =
  printf "@[<h>";
  for n = 1 to 40 do
    let n = Sint.of_int n in
    let x = (of_sint n) / 4. in
    printf "lngamma %.2f -> %.9f\n" x (lngamma x);
  done;

  let rec fn xs = begin
    match xs with
    | [] -> ()
    | x :: xs' -> begin
        printf "lngamma %.2f -> %.5e\n" x (lngamma x);
        fn xs'
      end
  end in
  fn [neg_inf; -1.; -0.; 0.; inf; nan];
  printf "@]"

let _ = test ()
