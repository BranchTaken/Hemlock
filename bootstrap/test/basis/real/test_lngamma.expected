lngamma 0.25 -> 1.288022525
lngamma 0.50 -> 0.572364943
lngamma 0.75 -> 0.203280951
lngamma 1.00 -> 0.000000000
lngamma 1.25 -> -0.098271836
lngamma 1.50 -> -0.120782238
lngamma 1.75 -> -0.084401121
lngamma 2.00 -> 0.000000000
lngamma 2.25 -> 0.124871715
lngamma 2.50 -> 0.284682870
lngamma 2.75 -> 0.475214667
lngamma 3.00 -> 0.693147181
lngamma 3.25 -> 0.935801931
lngamma 3.50 -> 1.200973602
lngamma 3.75 -> 1.486815579
lngamma 4.00 -> 1.791759469
lngamma 4.25 -> 2.114456927
lngamma 4.50 -> 2.453736571
lngamma 4.75 -> 2.808571419
lngamma 5.00 -> 3.178053830
lngamma 5.25 -> 3.561375910
lngamma 5.50 -> 3.957813968
lngamma 5.75 -> 4.366716037
lngamma 6.00 -> 4.787491743
lngamma 6.25 -> 5.219603987
lngamma 6.50 -> 5.662562060
lngamma 6.75 -> 6.115915891
lngamma 7.00 -> 6.579251212
lngamma 7.25 -> 7.052185451
lngamma 7.50 -> 7.534364237
lngamma 7.75 -> 8.025458396
lngamma 8.00 -> 8.525161361
lngamma 8.25 -> 9.033186920
lngamma 8.50 -> 9.549267257
lngamma 8.75 -> 10.073151240
lngamma 9.00 -> 10.604602903
lngamma 9.25 -> 11.143400120
lngamma 9.50 -> 11.689333421
lngamma 9.75 -> 12.242204940
lngamma 10.00 -> 12.801827480
lngamma -inf -> inf
lngamma -1.00 -> inf
lngamma -0.00 -> inf
lngamma 0.00 -> inf
lngamma inf -> inf
lngamma nan -> nan
